### DISCUSSION TITLE
We fetch successfully with certain single word query parameters but unsuccessfully when there is an underscore involved

### DISCUSSION THREAD

**Discussion by gregorsart** [neuefische/questions#770] [Web] [answered]
Hi,

as stated in the headline, we observed a strange behaviour when we fetch with queries like:

`good_with_other_dogs` or `good_with_children`

"Single words" like `trainability`, `barking` or `energy` work fine.

See attached screenshots: One shows how this "unsuccessful fetch" reflects in our react frontend render. (no dog name, but points with the value 10 in the list)

![Bildschirmfoto 2023-10-24 um 12 34 41](https://github.com/neuefische/questions/assets/32411569/381c51e6-9713-401f-93d4-8220d1b5e3e2)
![Bildschirmfoto 2023-10-24 um 12 47 42](https://github.com/neuefische/questions/assets/32411569/4b3c9924-c52a-44e6-b55c-444a04f83d72)

Is fetching with an underscore problematic or is our api not working properly?

Ty in advance

**Comment by gregorsart** [comment]
[Link to our PR](https://github.com/alinarub/dog-app/pull/13)

The story is completely coded, except for those two questions with underscores in the query parameter.

**Reply by NikRoe** [reply, marked as answer]
Hi there,

according to the documentation of the API, one of these parameters is required when making a request:

name
min_height 
max_height
min_weight
max_weight
min_life_expectancy
max_life_expectancy 
shedding
barking
energy
protectiveness
trainability 

I tried fetching dogs with the parameter `min_height` and got results.

Thus, the issue is not related to the underscore, but to invalid query parameters, since neither `good_with_children` or `good_with_other_dogs` is part of the list of acceptable parameters.

**Comment by gregorsart** [comment]
Hi Niklas,

ty for the fast response. 
You are right!

PS: We completely focused on the query results object and not on the possible query parameters.

We will continue tomorrow.
